File: 035 Search Insert Position/035 Search Insert Position.py
class Solution(object):
    def searchInsert(self, nums, target):
        if target <= nums[0]:
            return 0
        l = 0
        r = len(nums)
        m = Int((l+r)/2)
        if target > nums[r-1]:
            return r
        elif target == nums[r-1]:
            return r-1
        while True :
            if nums[m] >= target and nums[m-1] < target:
                return m
            elif nums[m] > target:
                r = m
            elif nums[m] < target:
                l = m
            m = Int((l+r)/2+0.5)
#Min:36ms

File: 035 Search Insert Position/035 Search Insert Position.c
int searchInsert(int* nums, int numsSize, int target) {
    if( *(nums) >= target )
        return 0;
    int l = 0, r = numsSize, m = (int) ((l+r)/2+0.5);
    if( target > *(nums+r-1) )
        return r;
    if(target == *(nums+r-1) )
        return r-1;
    while(true){
        if(target <= *(nums+m) && target > *(nums+m-1))
            return m;
        else if(target < *(nums+m))
            r = m;
        else if(target > *(nums+m))
            l = m;
        m = (int) ((l+r)/2+0.5);
    }
}
//Min:4ms